File: js/react-native/examples/event_builder.ts
import {
  EventBuilder,
  Keys,
  Kind,
  KindStandard,
  SingleThreadPow,
} from "../src/generated/nostr_sdk";

export function main(): string[] {
  const keys = Keys.generate();
  const kind = Kind.fromStd(KindStandard.TextNote);
  const textNote = new EventBuilder(
    kind,
    "Note from rust-nostr React Native bindings",
  ).finalize(keys);

  const customBuilder = new EventBuilder(new Kind(1234), "My custom content");
  const customEvent = customBuilder.finalize(keys);
  const powEvent = customBuilder
    .finalizeUnsigned(keys.publicKey())
    .mine(new SingleThreadPow(), 8)
    .sign(keys);
  const unsignedEvent = customBuilder.finalizeUnsigned(keys.publicKey());

  return [
    textNote.asJson(),
    customEvent.asJson(),
    powEvent.asJson(),
    unsignedEvent.asJson(),
  ];
}

File: swift/Examples/EventBuilder.swift
import NostrSDK

func eventBuilderExample() throws -> [String] {
    let keys = Keys.generate()
    let kind = Kind.fromStd(e: .textNote)

    let textNote = try EventBuilder(
        kind: kind,
        content: "Note from rust-nostr Swift bindings"
    ).finalize(signer: keys)

    let customBuilder = EventBuilder(
        kind: Kind(kind: 1234),
        content: "My custom content"
    )
    let customEvent = try customBuilder.finalize(signer: keys)
    let powEvent = try customBuilder
        .finalizeUnsigned(publicKey: keys.publicKey())
        .mine(adapter: SingleThreadPow(), difficulty: 8)
        .sign(signer: keys)
    let unsignedEvent = customBuilder.finalizeUnsigned(publicKey: keys.publicKey())

    return [
        textNote.asJson(),
        customEvent.asJson(),
        powEvent.asJson(),
        unsignedEvent.asJson(),
    ]
}
